src/Data/Either/Open.purs

```diff
@@ -0,0 +1,146 @@
+type Either10 a b c d e f g h i z = Either (Either9 a b c d e f g h i) z
+
+in1 :: forall z a. a -> Nested.Either2 z a
+in1 = Right
+
+in2 :: forall z a b. a -> Nested.Either3 z a b
+in2 v = Left (Right v)
+
+in3 :: forall z a b c. a -> Nested.Either4 z a b c
+in3 v = Left (Left (Right v))
+
+in4 :: forall z a b c d. a -> Nested.Either5 z a b c d
+in4 v = Left (Left (Left (Right v)))
+
+in5 :: forall z a b c d e. a -> Nested.Either6 z a b c d e
+in5 v = Left (Left (Left (Left (Right v))))
+
+in6 :: forall z a b c d e f. a -> Nested.Either7 z a b c d e f
+in6 v = Left (Left (Left (Left (Left (Right v)))))
+
+in7 :: forall z a b c d e f g. a -> Nested.Either8 z a b c d e f g
+in7 v = Left (Left (Left (Left (Left (Left (Right v))))))
+
+in8 :: forall z a b c d e f g h. a -> Nested.Either9 z a b c d e f g h
+in8 v = Left (Left (Left (Left (Left (Left (Left (Right v)))))))
+
+in9 :: forall z a b c d e f g h i. a -> Nested.Either10 z a b c d e f g h i
+in9 v = Left (Left (Left (Left (Left (Left (Left (Left (Right v))))))))
+
+in10 :: forall z a b c d e f g h i j. a -> Nested.Either10 (Either z a) b c d e f g h i j
+in10 v = Left (Left (Left (Left (Left (Left (Left (Left (Left (Right v)))))))))
+
+at1 :: forall z r a. r -> (a -> r) -> Nested.Either2 z a -> r
+at1 b a y = case y of
+  Right r -> a r
+  _ -> b
+
+at2 :: forall z r a b. r -> (a -> r) -> Nested.Either3 z a b -> r
+at2 b a y = case y of
+  Left (Right r) -> a r
+  _ -> b
+
+at3 :: forall z r a b c. r -> (a -> r) -> Nested.Either4 z a b c -> r
+at3 b a y = case y of
+  Left (Left (Right r)) -> a r
+  _ -> b
+
+at4 :: forall z r a b c d. r -> (a -> r) -> Nested.Either5 z a b c d -> r
+at4 b a y = case y of
+  Left (Left (Left (Right r))) -> a r
+  _ -> b
+
+at5 :: forall z r a b c d e. r -> (a -> r) -> Nested.Either6 z a b c d e -> r
+at5 b a y = case y of
+  Left (Left (Left (Left (Right r)))) -> a r
+  _ -> b
+
+at6 :: forall z r a b c d e f. r -> (a -> r) -> Nested.Either7 z a b c d e f -> r
+at6 b a y = case y of
+  Left (Left (Left (Left (Left (Right r))))) -> a r
+  _ -> b
+
+at7 :: forall z r a b c d e f g. r -> (a -> r) -> Nested.Either8 z a b c d e f g -> r
+at7 b a y = case y of
+  Left (Left (Left (Left (Left (Left (Right r)))))) -> a r
+  _ -> b
+
+at8 :: forall z r a b c d e f g h. r -> (a -> r) -> Nested.Either9 z a b c d e f g h -> r
+at8 b a y = case y of
+  Left (Left (Left (Left (Left (Left (Left (Right r))))))) -> a r
+  _ -> b
+
+at9 :: forall z r a b c d e f g h i. r -> (a -> r) -> Nested.Either10 z a b c d e f g h i -> r
+at9 b a y = case y of
+  Left (Left (Left (Left (Left (Left (Left (Left (Right r)))))))) -> a r
+  _ -> b
+
+at10 :: forall z r a b c d e f g h i j. r -> (a -> r) -> Nested.Either10 (Either z a) b c d e f g h i j -> r
+at10 b a y = case y of
+  Left (Left (Left (Left (Left (Left (Left (Left (Left (Right r))))))))) -> a r
+  _ -> b
+
+runEither1 :: forall a. Either1 a -> a
+runEither1 y = unsafePartial case y of
```

Comment:
You don't need to do this, you can match on `Left` and use `absurd` to go from `Void` to `a`.